### PR TITLE
Bump package version to recompile binaries with fixed gcc stack prote…

### PR DESCRIPTION
**Pull request by jslobodzian**
…ction (CVE-2023-4039)

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

### CODE REVIEW

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -28,7 +28,7 @@
 Summary:        Linux Kernel
 Name:           kernel
 Version:        5.15.131.1
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (mfrw) [inline SPECS/kernel/kernel.spec:31]: I think we might also want to bump the release in `kernel-headers` & `kernel-signed` to keep things in parity.

**Review by mfrw — COMMENTED** [review]
A few additional changes

File: SPECS/efibootmgr/efibootmgr.spec [hunk 2]
```diff
@@ -33,6 +33,9 @@ gzip -9 %{buildroot}%{_mandir}/man8/efibootdump.8
 %{_sbindir}/*
 %{_mandir}/man8/*
 %changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 16-4
```
Comment (anphel31) [inline SPECS/efibootmgr/efibootmgr.spec:36]: can you add a blank space before %changelog

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (jslobodzian) [reply SPECS/kernel/kernel.spec:31]: yes...build revealed that issue.  Done.

File: SPECS/ypbind/ypbind.spec [hunk 3]
```diff
@@ -502,7 +505,10 @@ install -m 755 %{SOURCE5} %{buildroot}%{_libexecdir}/ypbind-post-waitbind
 
 * Tue Oct 21 2008 Vitezslav Crhonek <vcrhonek@redhat.com> - 3:1.20.4-9
 - Merge Review - remove dot from end of the summary, convert all tags
-  in %%changelog to utf-8, escape %% character in changelog, fix
+  in %%changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 2.7.2-11
+- Recompile with stack-protection fixed gcc version (CVE-2023-4039)
+ to utf-8, escape %% character in changelog, fix
```
[the remark this answers is not in the document]
Comment (jslobodzian) [reply SPECS/ypbind/ypbind.spec:511]: fixed.

File: SPECS/openssh/openssh.spec [hunk 4]
```diff
@@ -263,6 +263,9 @@ fi
 %{_mandir}/man8/ssh-sk-helper.8.gz
 
 %changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 8.9p1-
```
[the remark this answers is not in the document]
Comment (jslobodzian) [reply SPECS/openssh/openssh.spec:266]: Fixed

File: SPECS/libwacom/libwacom.spec [hunk 5]
```diff
@@ -180,6 +183,9 @@ install -d %{buildroot}/%{_udevrulesdir}
 
 * Fri Feb 09 2018 Igor Gnatenko <ignatenkobrain@fedoraproject.org> - 0.28-2
 - Escape macros in %%changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 1.6-5
```
[the remark this answers is not in the document]
Comment (jslobodzian) [reply SPECS/libwacom/libwacom.spec:186]: Fixed.

File: SPECS/efibootmgr/efibootmgr.spec (hunk 2 above)
Comment (jslobodzian) [reply SPECS/efibootmgr/efibootmgr.spec:36]: Fixed

File: SPECS/qt5-qtdeclarative/qt5-qtdeclarative.spec [hunk 6]
```diff
@@ -141,6 +141,9 @@ popd
 
 
 %changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 5.12.5-
```
Comment (sameluch) [inline SPECS/qt5-qtdeclarative/qt5-qtdeclarative.spec:144]: Looks like we might be missing the release number change here from 4 to 5

File: SPECS/gcc/gcc.spec [hunk 7]
```diff
@@ -287,6 +287,9 @@ make %{?_smp_mflags} check-gcc
 %{_lib64dir}/libgomp.spec
 
 %changelog
+* Wed Sep 20 2023 Jon Slobodzian <joslobo@microsoft.com> - 11.2.0-6
```
Comment (anphel31) [inline SPECS/gcc/gcc.spec:290]: bumping gcc shouldn't be necessary

**Review by dmcilvaney — APPROVED** [review]
`OpenSSL`, `KeysInUse-OpenSSL` bits look ok to me.

**Review by gjswalling — APPROVED** [review]

**Review by sameluch — APPROVED** [review]

**Comment by anphel31** [comment]
Just adding a note about [CVE-2023-4039](https://nvd.nist.gov/vuln/detail/CVE-2023-4039). The gcc stack protection vulnerability is only applicable to ARM64 architecture; although we need to re-release X86_64 packages at the same time (no practical way to separate them). The patch was added to gcc in #6213

**Review by anphel31 — APPROVED** [review]

**Review by Camelron — APPROVED** [review]

**Review by rlmenge — APPROVED** [review]

**Pull request merged**